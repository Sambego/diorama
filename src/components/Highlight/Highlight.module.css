.highlight {
  padding: 0.5rem;
  background-color: var(--color-primary);
}
