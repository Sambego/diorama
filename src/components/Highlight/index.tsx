import React from "react";
import PropTypes from "prop-types";

import styles from "./Highlight.module.css";

type HighlightProps = {
	className?: string;
	style?: React.CSSProperties;
	color?: React.CSSProperties["backgroundColor"];
};
function Highlight({
	children,
	style,
	className,
	color,
}: React.PropsWithChildren<HighlightProps>) {
	return (
		<span
			style={{ backgroundColor: color, ...style }}
			className={`${styles.highlight} diorama-highlight ${className}`}
		>
			{children}
		</span>
	);
}

Highlight.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	color: PropTypes.string,
	style: PropTypes.shape({}),
};

Highlight.defaultProps = {
	className: "",
	color: undefined,
	style: {},
};

export default Highlight;
