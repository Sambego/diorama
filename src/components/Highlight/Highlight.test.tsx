import React from "react";
import renderer from "react-test-renderer";
import Highlight from ".";

it("renders correctly", () => {
	const tree = renderer
		.create(<Highlight>Highlighted text</Highlight>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
