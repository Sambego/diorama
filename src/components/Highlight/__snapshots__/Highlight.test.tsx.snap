// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<span
  className="_highlight_1eco9_1 diorama-highlight "
  style={
    {
      "backgroundColor": undefined,
    }
  }
>
  Highlighted text
</span>
`;
